source/basic.tex

```diff
@@ -4624,35 +4624,37 @@
 memory location.
 
 \pnum
-The library defines a number of atomic operations\iref{atomics} and
-operations on mutexes\iref{thread} that are specially identified as
-synchronization operations. These operations play a special role in making
-assignments in one thread visible to another. A synchronization operation on one
-or more memory locations is either a consume operation, an acquire operation, a
-release operation, or both an acquire and release operation. A synchronization
-operation without an associated memory location is a fence and can be either an
-acquire fence, a release fence, or both an acquire and release fence. In
-addition, there are relaxed atomic operations, which are not synchronization
-operations, and atomic read-modify-write operations, which have special
-characteristics. \begin{note} For example, a call that acquires a mutex will
+All modifications to a particular atomic object \placeholder{M} occur in some
+particular total order, called the \defn{modification order} of \placeholder{M}.
+\begin{note} There is a separate order for each
+atomic object. There is no requirement that these can be combined into a single
+total order for all objects. In general this will be impossible since different
+threads may observe modifications to different objects in inconsistent orders.
+\end{note}
+
+\pnum
+Certain library calls \defn{synchronize with} other library calls performed by
+another thread
+(\ref{support.signal}, \ref{util.smartptr.atomic}, \ref{atomics.order},
```

Comment:
I'd prefer the list of sections to be more clearly non-normative.

Reply:
Hm... I could move it down into the note after "The specifications of the synchronization operations", but I'd actually prefer to keep the list here, right at the top. A missed cross-reference is at most an editorial bug.

(Also, the library clauses should make sure they have a cross-reference to this section whenever they mention "synchronize with". This would then be a clear indicator that we mean the normative cross-thread meaning of "synchronize with". Similar to what we did with odr-use in the core section.)

Reply:
Are there other places where the standard, at the point where it defines a term, provides cross-references to uses of the term? I can't recall any, and it doesn't seem like a good practice- it would be hard to maintain, and the existence of the list would create the impression that the list is exhaustive (and thereby creating confusion around any use of "synchronize with" that's omitted from the list).

I'm fine with having uses of this term cross-reference the definition, of course, just not vice-versa.

I'd also be OK with making this list explicitly non-exhaustive, e.g. by starting it with "such as", although I still wouldn't understand why it's there.

Reply:
@geoffromer: The SFINAE rule lists the cases where it applies in a non-normative note [temp.deduct] p11.
I've moved the list of cross-references to the note; it still seems useful to the reader to have some cross-references to give an impression what those synchronization operations might be.

Reply:
Looks good.